### DISCUSSION TITLE
New options and features

### DISCUSSION THREAD

**Discussion by ShahriarKh** [leonhma/obsidian-functionplot#63] [General]
First of all, let me thank you for this useful obsidian plugin!

Since this plugin uses https://github.com/mauriciopoppe/function-plot under the hood, I think it will be easy to implement new options that are already supported by the "Function Plot". I took a look at the code, and since options are passed to `functionPlot`, seems like new ones can be added easily with some time & effort.

I really like to help contribute, so please tell me what you think and where should I start @leonhma.

---

List of some feature requests, already supported by "Function Plot":

- [ ] #44 
- [ ] #41

**Comment by leonhma** [comment]
Thank you for wanting to contribute to this project.

I just took a look at the website of function-plot, and holy ... A lot of new functionality has been added since I last viewed that page. I think that these features are quite reasonable to implement now, although you seem to know much more about typescript than I do (this was kind of my first typescript project 🫣). I think it'd be a great idea to have a dropdown for the function type maybe? There has to be a tradeoff between flexibility and ease of use somewhere... But I am open for suggestions and changes to the UI. You can create a new branch from main if you want and open a pull request, and I'll see if I can fix the jest and deepsource ci in the meantime...

**Reply by ShahriarKh** [reply]
It's a passion to help this plugin grow 😉 
Believe me or not, I haven't used TypeScript yet... but I can take a look at the source code and see where I can help.
For flexibility vs ease of use, I think the current UX is good. But for new options, we can follow the Function Plot syntax... maybe.
